tests/integration/test_ha.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Integration test for haproxy charm."""
+import requests
+from juju.application import Application
+
+from .conftest import get_unit_ip_address
+
+
+async def test_ha(application: Application, hacluster: Application):
+    """
+    arrange: deploy the chrony charm.
+    act: request chrony_exporter metrics endpoint.
+    assert: confirm that metrics are scraped.
+    """
+    await application.add_unit(count=3)
+    await application.model.wait_for_idle(
```

Comment:
I'd remove this since it might point us to race conditions when the async operations do not finish in the same order

Reply:
Even with `await`?

Reply:
I'm marking this as resolved for now, if you think we need to discuss about it more I'll move it to an internal thread!